fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/InitMaterializationContextHook.java

```diff
@@ -138,6 +140,16 @@ private List<MaterializationContext> createAsyncMaterializationContext(CascadesC
             MTMVCache mtmvCache = null;
             try {
                 mtmvCache = materializedView.getOrGenerateCache(cascadesContext.getConnectContext());
+                // If mv property use_for_rewrite is set false, should not partition in
+                // query rewrite by materialized view
+                String usedForRewrite = materializedView.getMvProperties().get(
+                        PropertyAnalyzer.PROPERTIES_USE_FOR_REWRITE);
+                if (!StringUtils.isEmpty(usedForRewrite) && !Boolean.parseBoolean(usedForRewrite)) {
+                    LOG.debug(String.format("mv doesn't part in query rewrite process because "
+                            + "use_for_rewrite is false, mv is %s",
+                            materializedView.getName()));
```

Comment:
avoid use string format, use '{}' in message template to avoid new string object for every log even if not print it

Reply:
Have fixed it